addons/mail/models/mail_message.py

```diff
@@ -659,9 +659,29 @@ def unlink(self):
         self.mapped('attachment_ids').filtered(
             lambda attach: attach.res_model == self._name and (attach.res_id in self.ids or attach.res_id == 0)
         ).unlink()
+        messages_by_partner = defaultdict(lambda: self.env['mail.message'])
         for elem in self:
+            for partner in elem.partner_ids:
+                messages_by_partner[partner] |= elem
             if elem.is_thread_message():
                 elem._invalidate_documents()
+
+        # Notify front-end of messages deletion for users using inbox
+        if messages_by_partner:
+            partner_id_involved = [partner.id for partner in messages_by_partner.keys()]
+            partner_involved_using_inbox = self.env['res.partner'].browse([
+                user['partner_id'][0]
+                for user in self.env['res.users'].search_read(
```

Comment:
Is that really necessary ? I mean, can't we send a notification for all internal partners even if they use email ? We could simply keep partners that are not ``partner_share`` (aka: linked to internal users). If we really have to target only users with notification type, why not benefiting from the cache and do something like
```
internal_users = partners.filtered(
    lamba partner: not partner.partner_share
).user_ids.filtered(
    lambda user: not user.share and user.notification_type == 'inbox'
)  # keep internal users even in multi-users mode
```

cc @seb-odoo

Reply:
Just filtering on partners that actually have a user is probably a very good first step, with minimal queries.

Checking for `partner_share` looks ok, even if we could imagine some day that why not portal users could benefit from something similar.

Checking for notification type looks indeed overkill, especially because it's also wrong, users that don't have inbox might also need to be notified if they starred the message or something like that. (the part about sending the notification to partner that starred the message is still lacking here by the way)

Reply:
Thanks for the explanations and the propositions. To keep it simple, I have then only checked that the partner has an user (4 queries are needed for that).